storage/readservice/store.go

```diff
@@ -67,18 +71,14 @@ func (s *store) ReadGroup(ctx context.Context, req *datatypes.ReadGroupRequest)
 	}
 
 	newCursor := func() (reads.SeriesCursor, error) {
-		cur, err := newIndexSeriesCursor(ctx, &source, req.Predicate, s.viewer)
-		if cur == nil || err != nil {
-			return nil, err
-		}
-		return cur, nil
+		return newIndexSeriesCursor(ctx, &source, req.Predicate, s.viewer)
 	}
 
 	return reads.NewGroupResultSet(ctx, req, newCursor), nil
 }
 
 func (s *store) TagKeys(ctx context.Context, req *datatypes.TagKeysRequest) (cursors.StringIterator, error) {
-	span, _ := tracing.StartSpanFromContext(ctx)
+	span, ctx := tracing.StartSpanFromContext(ctx)
```

Comment:
Should we not update these to return `tracing.LogError` wherever we define the tracer?